Day12/src/services/studentServices.ts

```diff
@@ -0,0 +1,319 @@
+
+            }
+
+        })
+
+    })
+
+}
+
+
+async function getSubjects() {
+
+    let query = `
+        SELECT * FROM school.subject;
+    `;
+    return new Promise((resolve, reject) => {
+        sql.query(query, (err: any, res: any) => {
+
+            if (err)
+                reject(err);
+            else {
+                resolve(res.rows);
+
+            }
+
+        })
+
+    })
+
+}
+
+async function getClasses() {
+
+    let query = `
+        SELECT * FROM school.class;
+    `;
+    return new Promise((resolve, reject) => {
+        sql.query(query, (err: any, res: any) => {
+
+            if (err)
+                reject(err);
+            else {
+
+                resolve(res.rows);
+
+            }
+
+        })
+
+    })
+
+}
+
+
+async function getStudentsByClassId(classId: string) {
+
+    let query = `
+     
+      SELECT s.sid,s.name,s.gender,s.phone,studies_in.classid
+      FROM school.student s
+      LEFT JOIN school.studies_in ON s.sid = studies_in.sid
+      WHERE classid= $1;
+
+    `;
+    return new Promise((resolve, reject) => {
+        sql.query(query, [classId], (err: any, res: any) => {
+
+            if (err)
+                reject(err);
+            else {
+                resolve(res.rows);
+
+            }
+
+        })
+
+    })
+
+}
+
+async function getStudentsBySubjectId(subjectId: string) {
+
+    let query = `
+      
+      SELECT s.sid,s.name,s.gender,s.phone,c.subjectname,c.subjectid
+      FROM school.student s
+      INNER JOIN school.studies_in ON s.sid = studies_in.sid
+      INNER JOIN (
+	     SELECT c.classid,sb.subjectname,sb.subjectid
+      FROM school.class c 
+      INNER  JOIN school.schedule s ON c.classid = s.classid 
+      INNER  JOIN school.subject sb ON s.subjectid = sb.subjectid
+      WHERE sb.subjectid = $1
+      ) c ON studies_in.classid = c.classid;
+
+    `;
+    return new Promise((resolve, reject) => {
+        sql.query(query, [subjectId], (err: any, res: any) => {
+
+            if (err)
+                reject(err);
+            else {
+                resolve(res.rows);
+
+            }
+
+        })
+
+    })
+
+}
+
+async function getStudentsByTeacherId(teacherId: string) {
+
+    let query = `
+      
+      SELECT s.sid,s.name,s.gender,s.phone  , t.name AS teachername , t.subjectname
+      FROM school.student s
+      INNER JOIN school.studies_in ON s.sid = studies_in.sid
+      INNER JOIN(
+	  SELECT c.classid , t.tid , t.name , t.subjectname , t.subjectid
+      FROM school.class c
+      INNER JOIN school.schedule s ON c.classid = s.classid
+      INNER JOIN(
+	  SELECT t.tid,t.name,sb.subjectname,sb.subjectid
+      FROM school.teacher t
+      LEFT  JOIN school.teacher_takes tt ON t.tid = tt.tid 
+      LEFT  JOIN school.subject sb ON tt.subjectid = sb.subjectid
+      WHERE t.tid = $1
+      ) t ON s.subjectid = t.subjectid
+	  ) t ON studies_in.classid = t.classid;
+
+
+    `;
+    return new Promise((resolve, reject) => {
```

Comment:
same here, sql.query return promise we can just return that, if we want to manipulate data returned from sql.query, we can use .then and return updated data